File: SpringServer/spring/src/main/java/com/springbackend/service/RoutineService.java
package com.springbackend.service;

import com.springbackend.dto.RoutineDTO;
import com.springbackend.entity.Routine;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface RoutineService {
    RoutineDTO getRoutine(String routineName);
    void saveRoutine(RoutineDTO routineDTO);

    List<RoutineDTO> getAllRoutinesByUser(String userId);
}

File: CapAndroid/android/app/src/main/java/com/team6/routineapp/service/RoutineService.kt
package com.team6.routineapp.service

import com.team6.routineapp.dto.RoutineDTO
import retrofit2.Call
import retrofit2.http.*

interface RoutineService {
    @POST("routine/save")
    fun saveRoutine(@Body routineDTO: RoutineDTO): Call<Void>

    @GET("routine/{routineName}")
    fun getRoutine(@Path("routineName") routineName: String): Call<RoutineDTO>

    @GET("routine/all/{userId}")
    fun getAllRoutinesByUser(@Path("userId") userId: String): Call<List<RoutineDTO>>
}